src/pretty.rs

```diff
@@ -60,6 +60,17 @@ where
         self.text(s)
     }
 
+    fn multiline_string(&'a self, s: &str) -> DocBuilder<'a, Self, A> {
+        let percent_count = s.split(|c| c != '%').map(|v| v.len()).max().unwrap();
+        let delimiter = "%".repeat(percent_count + 1);
```

Comment:
This might be too clever, but I like it 🙂

Reply:
FWIW, I think it's fine :slightly_smiling_face: 

It's not the best we can do (here we count any percent sign, while for example a simple modulo alone doesn't require double delimiters). A step further would be to match on sequences we know the lexer would interpret, i.e `m%"`, `"%`, and `%{`. But at least it's correct, and it's probably ok for now. Maybe add a comment to say so? (that the counts is over-approximated)

Reply:
Actually, that was a terrible idea. We have a function `min_interpolate_sign` in the same file to compute the precise count needed 🤦